### ISSUE TITLE
Add a process to a manufacturing work order

### ISSUE THREAD

**Issue by bhaugen** [valnet/valuenetwork#444]
You can now add processes to workflow orders, but not manufacturing ones.

Workflow orders take the same resource through processes of change, where the same resource is both an input and an output. A workflow order is basically a linear flow.

Manufacturing orders assemble different input resources into different output resources. A manufacturing order is typically a tree, but can have parallel processes, which can make it into a network: a directed graph. 

To add a process to the beginning of a manufacturing order, the process must produce a resource that is an input to one of the starting processes of the order: either because that starting process already required a resource of that type, or by adding one.

To add a process to the end of a manufacturing order, the process must take the output resource of one of the end processes of the order as an input.  Moreover, that output process had been an order item, to the output of the new process must become an order item, replacing the previous one.

Inserting a process between some other processes in the order requires taking the output of the previous process as an input to the new process and replacing the input of the next process with an output of the new process.

The best way to do any of those additions or insertions would be visually. See [visual language experiments](https://docs.google.com/presentation/d/1N81Q0uDjA7OMEZTLNig6O-MLuWHgb5TL0IxPZ2U8qDU/edit?usp=sharing). But it would be possible to do it more awkwardly with forms, as done now in workflow orders.

**Comment by fdurville** [comment]
I started to look at the presentation on the visual language. Are you familiar with LabView? 
FD

**Comment by bhaugen** [comment]
Yes, used LabView once a long time ago. I've also used several other visual languages where you wire visual components together. Having a conversation with Conway of Conway's Law, who is designing a new one. They have some drawbacks that you need to avoid, like too many arrows that crisscross all over the place. At which point, you would be better off with text.
